modules/add-contextual-data/context-info-db.c

```diff
@@ -260,19 +260,37 @@ context_info_db_get_selectors(ContextInfoDB *self)
   return g_hash_table_get_keys(self->index);
 }
 
+static void
+_truncate_eol(gchar *line, gsize line_len)
+{
+  if (line[line_len - 2] == '\r')
+    line[line_len - 2] = '\0';
+  else if (line[line_len - 1] == '\n')
+    line[line_len - 1] = '\0';
```

Comment:
Happens.

And I take the third point back. `line_buf_len` is good as is - `realloc()` took care of it. Though it might be useful to return `line_len` as well if we started supporting binary data later on or started optimizing away `strlen()'s`, that's what I tried to refer to. Binary data may not be that useful for a CSV, but other input forms might support that better.